lib/core/src/lib.rs

```diff
@@ -233,77 +234,96 @@ mod example_01 {
+            let fonts             = &mut world.fonts;
+            let font_id           = fonts.load_embedded_font("DejaVuSansMono").unwrap();
+
+            let mut text_component = crate::text::TextComponentBuilder{workspace,fonts,font_id,
+                text       : "".to_string(),
+                properties : TextComponentProperties {
+                    position  : Point2::new(-0.95,-0.9),
+                    size      : Vector2::new(1.8,1.6),
+                    text_size : 0.032,
+                    color     : Color {r: 0.0, g: 0.8, b: 0.0, a: 1.0},
+                }
+            }.build();
+            text_component.cursors.add_cursor(CharPosition{line:0,column:0});
+            workspace.text_components.push(text_component);
             world.workspace_dirty.set(workspace_id);
 
+            let now             = js_sys::Date::now();
+            let animation_start = now + 3000.0;
+            let start_scrolling = animation_start + 10000.0;
+            let mut chars       = typed_character_list(animation_start,include_str!("lib.rs"));
             world.on_frame(move |w| {
-                let space = &mut w.workspaces[workspace_id];
-                for text_component in &mut space.text_components {
-                    text_component.scroll(Vector2::new(0.0,0.00001));
-                }
-                w.workspace_dirty.set(workspace_id);
+                animate_text_component(w,workspace_id,&mut chars,start_scrolling)
             }).forget();
         });
     }
+
+    struct CharToPush {
+        time   : f64,
+        a_char: char,
+    }
+
+    const ONE_CHAR_TYPING_DURATION_MS : f64 = 50.0;
+
+    fn typed_character_list(start_time:f64, text:&'static str) -> Vec<CharToPush> {
+        text.char_indices().map(|(i,a_char)| {
+            let time = start_time + ONE_CHAR_TYPING_DURATION_MS * i as f64;
+            CharToPush {time,a_char}
+        }).collect()
+    }
+
+    fn animate_text_component
+    ( world:&mut World
+    , workspace_id:WorkspaceID
+    , typed_chars:&mut Vec<CharToPush>
+    , start_scrolling:f64) {
+        let workspace   = &mut world.workspaces[workspace_id];
+        let editor      = workspace.text_components.first_mut().unwrap();
+        let now         = js_sys::Date::now();
+
+        let to_type_now = typed_chars.drain_filter(|ch| ch.time <= now);
+        for ch in to_type_now {
+            let cursor = editor.cursors.cursors.first_mut().unwrap();
+            let string = ch.a_char.to_string();
+            let change = TextChange::insert(cursor.position, string.as_str());
+            editor.content.make_change(change);
+            let new_cursor_position = CharPosition {
+                line: editor.content.lines.len()-1,
+                column : editor.content.lines.last().unwrap().len(),
+            };
+            cursor.position = new_cursor_position;
+        }
+        if start_scrolling <= js_sys::Date::now() {
+            editor.scroll(Vector2::new(0.0, -0.01));
```

Comment:
This `y` scroll value seems very small. I remember looking at the demo when the text was "jumping up" several times per second. I understand that its `y` movement is encoded here. My question is if the `y` and `x` position are aligned with `y` and `x` pixels respectivelly? Is moving `2` to right equal to moving 2 pixels rights? I would like to talk about it also in context of our camera - probably this should be part of integration task the next sprint.

Reply:
So, some answers are in docs for `scroll` function used here. So: No, they aren't aligned with pixels. the 1.0 is equal to line's height (for both dimensions).

Reply:
Ok, we need to talk about it on Monday when we will be planning merging it with our camera system.